sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -2295,18 +2297,17 @@ def get_blob_tags(self, **kwargs):
         options = _get_blob_tags_options(version_id=version_id, snapshot=self.snapshot, **kwargs)
         try:
             _, tags = self._client.blob.get_tags(**options)
-            return parse_tags(tags) # pylint: disable=protected-access
+            return parse_tags(tags)   # type: ignore  # pylint: disable=protected-access
```

Comment:
Properly type `parse_tags`

Reply:
For some reason this is the typing in [main](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-blob/azure/storage/blob/_deserialize.py#L175):

`def parse_tags(generated_tags: Optional[List["BlobTag"]]) -> Union[Dict[str, str], None]:`

So looking for suggestions to how we can tell this method that yes we provided the optional thing and if you provide it -> guaranteed to be the `Dict[str, str` ret and not `None`

Reply:
Test get_blob_tags on blob with no tags.

Reply:
Okay so looking at the function `parse_tags`:
![image](https://github.com/Azure/azure-sdk-for-python/assets/101599632/b10347ca-a04d-4a09-bec1-46d24d53fa31)
I don't think this will ever return `None` because even on a Blob with 0 tags like I tested here, the thing it is conditioning on is `generated_tags` which will come back regardless I believe as long as the Blob exists.

So current behavior is returning an empty `{}` so I will consider this a bug and fix the code to properly type it.

Reply:
I lied, reverted. Explanation in `_deserialize` PR